test/extended/util/framework.go

```diff
@@ -238,10 +238,81 @@ func processScanError(log string) error {
 	return fmt.Errorf(log)
 }
 
+// getImageStreamObj returns the updated spec for imageStream object
+func getImageStreamObj(imageStreamName, imageRef string) *imagev1.ImageStream {
+	imageStream := &imagev1.ImageStream{
+		ObjectMeta: metav1.ObjectMeta{Name: imageStreamName},
+		Spec: imagev1.ImageStreamSpec{
+			LookupPolicy:          imagev1.ImageLookupPolicy{Local: true},
+			DockerImageRepository: imageRef,
+			Tags: []imagev1.TagReference{{
+				Name: "latest",
+				ImportPolicy: imagev1.TagImportPolicy{
+					ImportMode: imagev1.ImportModePreserveOriginal,
+				},
+			}},
+		},
+		Status: imagev1.ImageStreamStatus{
+			DockerImageRepository: imageRef,
+			Tags: []imagev1.NamedTagEventList{{
+				Tag: "latest",
+			}},
+		},
+	}
+	return imageStream
+}
+
+// WaitForImageStreams creates & waits for custom ruby imageStream to be available in current namespace
+// TODO: To eliminate the dependency on OpenShift Samples Operator in future,
+// WaitForImageStreams should be a replacement of WaitForOpenShiftNamespaceImageStreams func
+func WaitForImageStreams(oc *CLI) error {
```

Comment:
nit (won't block on this). We should consider adding the imagestream name and image ref as parameter. Don't need to do this now since this is refactoring previous code, but is worth considering as the Samples Operator becomes "off by default."

Ex: 

```suggestion
func WaitForImageStreamImport(oc *CLI, imageStreamName string, imageRef string) error {
```

Reply:
How about introducing this change while we implement the same logic for rest of the images too? At present [WaitForOpenShiftNamespaceImageStreams](https://github.com/openshift/origin/blob/master/test/extended/util/framework.go#L244) is still being used to import other imagestreams (as we have focused just on ruby atm).